support/cas-server-support-dynamodb-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/DynamoDbTicketRegistryFacilitatorBillingModePayPerRequestTests.java

```diff
@@ -0,0 +1,102 @@
+import org.apereo.cas.config.DynamoDbTicketRegistryConfiguration;
+import org.apereo.cas.config.DynamoDbTicketRegistryTicketCatalogConfiguration;
+import org.apereo.cas.config.support.CasWebApplicationServiceFactoryConfiguration;
+import org.apereo.cas.logout.config.CasCoreLogoutConfiguration;
+import org.apereo.cas.util.junit.EnabledIfPortOpen;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.autoconfigure.RefreshAutoConfiguration;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.BillingMode;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableRequest;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableResponse;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * This is {@link DynamoDbTicketRegistryFacilitatorBillingModePayPerRequestTests}.
+ *
+ * @author Ned Regina
+ *
+ */
+@EnabledIfPortOpen(port = 8000)
+@SpringBootTest(classes = {
+        DynamoDbTicketRegistryConfiguration.class,
+        DynamoDbTicketRegistryTicketCatalogConfiguration.class,
+        CasCoreTicketsConfiguration.class,
+        CasCoreTicketCatalogConfiguration.class,
+        CasCoreLogoutConfiguration.class,
+        CasCoreHttpConfiguration.class,
+        CasCoreServicesConfiguration.class,
+        CasCoreAuthenticationConfiguration.class,
+        CasCoreTicketIdGeneratorsConfiguration.class,
+        CasCoreServicesAuthenticationConfiguration.class,
+        CasCoreConfiguration.class,
+        CasCoreNotificationsConfiguration.class,
+        CasCoreWebConfiguration.class,
+        CasCoreUtilConfiguration.class,
+        CasWebApplicationServiceFactoryConfiguration.class,
+        CasCoreAuthenticationServiceSelectionStrategyConfiguration.class,
+        CasCoreAuthenticationHandlersConfiguration.class,
+        CasCoreAuthenticationMetadataConfiguration.class,
+        CasCoreAuthenticationPolicyConfiguration.class,
+        CasCoreAuthenticationPrincipalConfiguration.class,
+        CasCoreAuthenticationSupportConfiguration.class,
+        CasPersonDirectoryConfiguration.class,
+        RefreshAutoConfiguration.class
+},
+        properties = {
+                "cas.ticket.registry.dynamo-db.endpoint=http://localhost:8000",
+                "cas.ticket.registry.dynamo-db.drop-tables-on-startup=true",
+                "cas.ticket.registry.dynamo-db.local-instance=true",
+                "cas.ticket.registry.dynamo-db.region=us-east-1",
+                "cas.ticket.registry.dynamo-db.billingMode=PAY_PER_REQUEST"
+        })
+@Tag("DynamoDb")
+public class DynamoDbTicketRegistryFacilitatorBillingModePayPerRequestTests {
```

Comment:
Remove test class and reuse the same test class that exists. Add your changes there to verify the default billing mode.

Reply:
I used a separate test class because the configuration for the autowired dynamoDbTicketRegistryFacilitator is different between the two tests.  Since this config is set at the time of bean instantiation, I'd have to use reflection to update it between methods within the same test class.

Reply:
Ah, in that case, you should likely be using @Nested tests.

Reply:
For an example, please see `PrincipalAttributeRepositoryFetcherLdapTests`.

Since the properties control constructions of components and beans, what should be done is: you keep the configuration classes in one area to avoid duplication, and simply specify properties differently for each nested test. Different settings lead to different autowiring behavior, and easier maintenance in the long-run since we'd not be duplicating 20 config classes yet again to test a single line.

Reply:
Perfect!  Thanks for the pointer.